File: js/tscripts/Hdr.js
if (loglvl) print("--> Hdr.js");

var RPMTAG_NAME = 1000;
var RPMTAG_REQUIRENAME = 1049;
var RPMTAG_BASENAMES = 1117;

var N = "popt";

var ts = new Ts();
// var mi = ts.mi(RPMTAG_NAME, 'popt');
var mi = new Mi(ts, RPMTAG_NAME, N);
ack("mi.length", 1);
ack("mi.count", 1);
ack("mi.instance", 0);  // zero before iterating

// var h = new Hdr();
for (let [dbkey,h] in Iterator(mi)) {
    nack("mi.instance", 0); // non-zero when iterating

    ack("typeof h;", "object");
    ack("h instanceof Hdr;", true);
    ack("h.debug = 1;", 1);
    ack("h.debug = 0;", 0);

// for (var [tagno,tagval] in Iterator(h,true))
//     print(h.tag(tagno),JSON.stringify(tagval));

    var ds = h.ds();
    ack("typeof ds;", "object");
    ack("ds instanceof Ds;", true);
    var ds = h.ds(RPMTAG_NAME);
    ack("typeof ds;", "object");
    ack("ds instanceof Ds;", true);
    var ds = h.ds(RPMTAG_REQUIRENAME);
    ack("typeof ds;", "object");
    ack("ds instanceof Ds;", true);

    var fi = h.fi();
    ack("typeof fi;", "object");
    ack("fi instanceof Fi;", true);
    var fi = h.fi(RPMTAG_BASENAMES);
    ack("typeof fi;", "object");
    ack("fi instanceof Fi;", true);

    ack("h.name", N);
    ack("h[RPMTAG_NAME]", N);
//  ack("h['name']", N);	// not yet ...
    ack("h.epoch", undefined);
    ack("h.version", undefined);
    ack("h.release", undefined);

    ack("h.summary", undefined);
    ack("h.description", undefined);
    ack("h.buildtime", undefined);
    ack("h.buildhost", undefined);
    ack("h.installtime", undefined);
    ack("h.size", undefined);
    ack("h.distribution", undefined);
    ack("h.vendor", undefined);
    ack("h.license", undefined);
    ack("h.packager", undefined);
    ack("h.group", undefined);

    ack("h.changelogtime", undefined);
    ack("h.changelogname", undefined);

    ack("h.url", undefined);
    ack("h.os", undefined);
    ack("h.arch", undefined);

    ack("h.prein", undefined);
    ack("h.preinprog", undefined);
    ack("h.postin", undefined);
    ack("h.postinprog", undefined);
    ack("h.preun", undefined);
    ack("h.preunprog", undefined);
    ack("h.postun", undefined);
    ack("h.postunprog", undefined);

    ack("h.filenames", undefined);
    ack("h.dirnames", undefined);
    ack("h.basenames", undefined);

    ack("h.filesizes", undefined);
    ack("h.filestates", undefined);
    ack("h.filemodes", undefined);
    ack("h.filerdevs", undefined);

    ack("h.sourcerpm", undefined);
    ack("h.archivesize", undefined);

    ack("h.providename", undefined);
    ack("h.requirename", undefined);
    ack("h.conflictname", undefined);
    ack("h.obsoletesname", undefined);

//  ack("h.keys()", undefined);
//  var legacyHeader = true;
//  ack("h.unload(legacyHeader)", undefined);
    var origin = "http://rpm5.org/files/popt/popt-1.14-1.i386.rpm";
    ack("h.setorigin(origin)", origin);
    ack("h.getorigin()", origin);
    var qfmt = "%{buildtime:date}";
    ack("h.sprintf(qfmt)", undefined);
}

delete mi;	// GCZeal?
delete ts;	// GCZeal?

if (loglvl) print("<-- Hdr.js");

File: ruby/tscripts/Hdr.rb
if ($loglvl)
  print("--> Hdr.rb\n");
end

$RPMTAG_NAME = 1000;
$RPMTAG_REQUIRENAME = 1049;
$RPMTAG_BASENAMES = 1117;

$N = "popt";

$ts = Ts.new();
## $mi = $ts.mi($RPMTAG_NAME, $N);
#$mi = Mi.new($ts, $RPMTAG_NAME, $NN);
#ack("$mi.instance_of?(Hdr)", true)
#ack("$mi.kind_of?(Hdr)", true)
#ack("$mi.class.to_s", "Hdr")
#ack("$mi.length", 1);
#ack("$mi.count", 1);
#ack("$mi.instance", 0);  # zero before iterating

$h = Hdr.new();
# $h = $mi.next()
## nack("$mi.instance", 0); # non-zero when iterating

#$ds = $h.ds();
#ack("$ds.instance_of?(Ds)", true)
#ack("$ds.kind_of?(Ds)", true)
#ack("$ds.class.to_s", "Ds")

#$ds = $h.ds($RPMTAG_NAME);
#ack("$ds.instance_of?(Ds)", true)
#ack("$ds.kind_of?(Ds)", true)
#ack("$ds.class.to_s", "Ds")

#$ds = $h.ds($RPMTAG_REQUIRENAME);
#ack("$ds.instance_of?(Ds)", true)
#ack("$ds.kind_of?(Ds)", true)
#ack("$ds.class.to_s", "Ds")

#$fi = $h.fi();
#ack("$fi.instance_of?(Fi)", true)
#ack("$fi.kind_of?(Fi)", true)
#ack("$fi.class.to_s", "Fi")

#$fi = $h.fi($RPMTAG_BASENAMES);
#ack("$fi.instance_of?(Fi)", true)
#ack("$fi.kind_of?(Fi)", true)
#ack("$fi.class.to_s", "Fi")

ack("$h['name']", $N);
ack("$h['epoch']", nil);
ack("$h['version']", nil);
ack("$h['release']", nil);
ack("$h['summary']", nil);
ack("$h['description']", nil);
ack("$h['buildtime']", nil);
ack("$h['buildhost']", nil);
ack("$h['installtime']", nil);
ack("$h['size']", nil);
ack("$h['distribution']", nil);
ack("$h['vendor']", nil);
ack("$h['license']", nil);
ack("$h['packager']", nil);
ack("$h['group']", nil);

ack("$h['changelogtime']", nil);
ack("$h['changelogname']", nil);

ack("$h['url']", nil);
ack("$h['os']", nil);
ack("$h['arch']", nil);

ack("$h['prein']", nil);
ack("$h['preinprog']", nil);
ack("$h['postin']", nil);
ack("$h['postinprog']", nil);
ack("$h['preun']", nil);
ack("$h['preunprog']", nil);
ack("$h['postun']", nil);
ack("$h['postunprog']", nil);

# ack("$h['filenames']", nil);
ack("$h['dirnames']", nil);
ack("$h['basenames']", nil);

ack("$h['filesizes']", nil);
ack("$h['filestates']", nil);
ack("$h['filemodes']", nil);
ack("$h['filerdevs']", nil);

ack("$h['sourcerpm']", nil);
ack("$h['archivesize']", nil);

ack("$h['providename']", nil);
ack("$h['requirename']", nil);
ack("$h['conflictname']", nil);
ack("$h['obsoletesname']", nil);

ack("$h.keys()", nil);
$legacyHeader = true;
ack("$h.unload($legacyHeader)", nil);
$origin = "http://rpm5.org/files/popt/popt-1.14-1.i386.rpm";
ack("$h.setorigin($origin)", $origin);
ack("$h.getorigin()", $origin);
$qfmt = "%{buildtime:date}";
ack("$h.sprintf($qfmt)", nil);

if ($loglvl)
  print("<-- Hdr.rb\n");
end
